scipy/interpolate/_aaa.py

```diff
@@ -0,0 +1,363 @@
+    >>> z = np.exp(np.linspace(-0.5, 0.5 + 15j*np.pi, 1000))
+    >>> r = AAA(lambda z: np.tan(np.pi*z/2), z)
+
+    We see that AAA takes 12 steps to converge with the following errors:
+    
+    >>> r.errors.size
+    12
+    >>> r.errors
+    array([2.49261500e+01+0.j, 4.28045609e+01+0.j, 1.71346935e+01+0.j,
+        8.65055336e-02+0.j, 1.27106444e-02+0.j, 9.90889874e-04+0.j,
+        5.86910543e-05+0.j, 1.28735561e-06+0.j, 3.57007532e-08+0.j,
+        6.37019486e-10+0.j, 1.67119305e-11+0.j, 1.17128023e-13+0.j])
+
+    We can also plot the computed poles:
+    
+    >>> fig, ax = plt.subplots()
+    >>> ax.plot(z.real, z.imag, '.', markersize=2, label="Sample points")
+    >>> ax.plot(r.poles.real, r.poles.imag, '.', markersize=5, label="Computed poles")
+    >>> ax.set(xlim=[-3.5, 3.5], ylim=[-3.5, 3.5], aspect="equal")
+    >>> ax.legend()
+    >>> plt.show()
+    """
+    def __init__(self, f, z, *, rtol=1e-13, max_terms=100):
+        # input validation
+        z = np.asarray(z).ravel()
+        if callable(f):
+            f = f(z)
+        else:
+            f = np.asarray(f).ravel()
+
+            if f.size != z.size:
+                raise ValueError("`f` and `z` must be the same size.")
+
+        # Remove infinite or NaN function values and repeated entries
+        to_keep = (np.isfinite(f)) & (~np.isnan(f))
+        f = f[to_keep]
+        z = z[to_keep]
+        z, uni = np.unique(z, return_index=True)
+        f = f[uni]
+
+        # Initialization for AAA iteration
+        M = np.size(z)
+        atol = rtol * np.linalg.norm(f, ord=np.inf)
+        mask = np.ones(M, dtype=np.bool_)
+        zj = np.empty(max_terms, dtype=np.complex128)
+        fj = np.empty(max_terms, dtype=np.complex128)
+        # Cauchy matrix
+        C = np.empty((M, max_terms), dtype=np.complex128)
+        # Loewner matrix
+        A = np.empty((M, max_terms), dtype=np.complex128)
+        errvec = np.empty(max_terms, dtype=np.complex128)
+        R = np.repeat(np.mean(f), M)
+
+        # AAA iteration
+        for m in range(max_terms):
+            # Introduce next support point
+            # Select next support point
+            jj = np.argmax(np.abs(f[mask] - R[mask]))
+            # Update support points
+            zj[m] = z[mask][jj]
+            # Update data values
+            fj[m] = f[mask][jj]
+            # Next column of Cauchy matrix
+            with np.errstate(divide="ignore", invalid="ignore"):
+                C[:, m] = 1 / (z - z[mask][jj])
+            # Update mask
+            mask[np.flatnonzero(mask)[jj]] = False
+            # Update Loewner matrix
+            with np.errstate(invalid="ignore"):
+                A[:, m] = (f - fj[m]) * C[:, m]
+
+            # Compute weights
+            rows = mask.sum()
+            if rows >= m + 1:
+                # The usual tall-skinny case
+                # Reduced SVD, use ?gesvd rather than ?gesdd to avoid convergence issues
+                _, s, V = scipy.linalg.svd(
```

Comment:
Same comment about NumPy svd versus SciPy svd.

Reply:
Numpy does not allow you use `gesvd` instead of `gesdd` which I have found to cause issues with convergence